l1-contracts/test/decoders/Decoder.t.sol

```diff
@@ -80,27 +50,21 @@ contract DecoderTest is DecoderBase {
 
     // Using the FULL decoder.
     (
-      uint256 l2BlockNumber,
-      bytes32 startStateHash,
-      bytes32 endStateHash,
-      bytes32 publicInputsHash,
+      bytes32 diffRoot,
+      bytes32 l1ToL2MessagesHash,
       bytes32[] memory l2ToL1Msgs,
       bytes32[] memory l1ToL2Msgs
-    ) = helper.decode(data.block.body);
-    (bytes32 diffRoot, bytes32 l1ToL2MessagesHash) =
-      helper.computeDiffRootAndMessagesHash(data.block.body);
+    ) = helper.computeConsumables(data.block.body);
 
     // Header
     {
-      (uint256 headerL2BlockNumber, bytes32 headerStartStateHash, bytes32 headerEndStateHash) =
-        headerHelper.decode(data.block.body);
-
-      assertEq(l2BlockNumber, data.block.blockNumber, "Invalid block number");
-      assertEq(headerL2BlockNumber, data.block.blockNumber, "Invalid block number");
-      assertEq(startStateHash, data.block.startStateHash, "Invalid start state hash");
-      assertEq(headerStartStateHash, data.block.startStateHash, "Invalid start state hash");
-      assertEq(endStateHash, data.block.endStateHash, "Invalid end state hash");
-      assertEq(headerEndStateHash, data.block.endStateHash, "Invalid end state hash");
+      HeaderDecoder.Header memory header = headerHelper.decode(data.block.header);
```

Comment:
When decoding the full header here it would probably be nice to match all the values, e.g., extend the json such that it checks every single value. That way it is instantly clear if header is decoded incorrectly.

Reply:
Good point. Addressed in https://github.com/AztecProtocol/aztec-packages/pull/4015/commits/b6bb599503f59f9740d404ce24a8d13d01e75db2